frontend/src/components/ScheduleSearch.tsx

```diff
@@ -46,9 +48,13 @@ const CourseCombobox = ({
   }, [activeSchedule]);
 
   function getCourses() {
+    // hyphenates 3 to 5 digit numbers, e.g. 152 -> 15-2, 1521 -> 15-21, 15213 -> 15-213
+    // otherwise, input value remains the same
+    const hyphenated = inputValue.replace(unhyphenatedCourseCodeRegex, "$1-$2");
     return allCourses.filter(
       (course) =>
         (course.courseID.includes(inputValue) ||
+          course.courseID.includes(hyphenated) ||
```

Comment:
In this case I think you don't even need the `.includes(inputValue)`. Otherwise LGTM.

Reply:
Resolved in ead2654